### PR TITLE
feat(component): add header tooltip for worksheet component

### PR DESCRIPTION
**Pull request by bc-apostoliuk**
## What?

Added header tooltip for worksheet component.

Refactored rendering of the tooltip for the Table component.
(packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx)

## Why?

To be able to leave a short description of the column.

## Screenshots/Screen Recordings
Screenshot 2023-01-11 at 12 16 27

## Testing/Proof

Locally + UT.

### CODE REVIEW

File: packages/big-design/src/components/Worksheet/spec.tsx [hunk 1]
```diff
@@ -1223,3 +1224,25 @@ describe('column widths', () => {
...
+
+    expect(getByTitle('Hover or focus for additional context.')).toBeTruthy();
+  });
+
+  test('renders tooltip when hovering on icon', async () => {
+    const { getByTitle } = render(
+      <Worksheet columns={columns} items={items} onChange={handleChange} />,
+    );
+
+    fireEvent.mouseOver(getByTitle('Hover or focus for additional context.'));
```
Comment (golcinho) [inline packages/big-design/src/components/Worksheet/spec.tsx:1242]:
```suggestion
    await userEvent.hover(getByTitle('Hover or focus for additional context.'));
```

File: packages/big-design/src/components/Worksheet/spec.tsx [hunk 2]
```diff
@@ -1223,3 +1224,25 @@ describe('column widths', () => {
...
+
+    const result = await screen.findByText('Info about number field');
```
Comment (golcinho) [inline packages/big-design/src/components/Worksheet/spec.tsx:1244]:
```suggestion
    const result = screen.getByText('Info about number field');
```

File: packages/big-design/src/components/Worksheet/spec.tsx (hunk 1 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Worksheet/spec.tsx]: Done

File: packages/big-design/src/components/Worksheet/spec.tsx (hunk 2 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Worksheet/spec.tsx]: Done

**Review by golcinho — APPROVED** [review]

**Review by eugene-polev — APPROVED** [review]

**Review by bc-andreadao — APPROVED** [review]

File: packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx [hunk 3]
```diff
@@ -66,29 +66,28 @@ const InternalHeaderCell = <T extends TableItem>({
     );
   };
 
-  // TODO: Update this. Return early if we don't have tooltip.
   const renderTooltip = () => {
-    if (typeof tooltip === 'string' && tooltip.length > 0) {
```
Comment (jorgemoya) [inline packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx:71]: Should we keep this check?

**Review by jorgemoya — COMMENTED** [review]
Looking good! One thing is missing is to add the `tooltip` prop to the column types in the docs.

File: packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx (hunk 3 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx:71]: @jorgemoya 
Thanks for the comment 😊.
I don't think so. I guess the checking like `if(!tooltip) { return null }` looks completely sufficient. Why do you think we need to keep the previous condition? `typeof tooltip === 'string' && tooltip.length > 0`

**Comment by bc-apostoliuk** [comment]
> Looking good! One thing is missing is to add the `tooltip` prop to the column types in the docs.

@jorgemoya Thanks, done 👍

File: packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx (hunk 3 above)
Comment (chanceaclark) [reply packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx:71]: It's specifically for library consumers that don't use TypeScript. It adds a layer of protection in the component and I recommend keeping it and using type-guards for other components.

Comment (bc-apostoliuk) [reply packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx:71]: @jorgemoya @chanceaclark 👍 It makes sense. I've returned to the previous check.
Thanks)

**Review by jorgemoya — APPROVED** [review]

File: packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx [hunk 4]
```diff
@@ -66,29 +66,28 @@ const InternalHeaderCell = <T extends TableItem>({
...
-                size="medium"
-                title="Hover or focus for additional context."
-              />
-            </Box>
-          }
-        >
-          {tooltip}
-        </Tooltip>
-      );
+    if (!(typeof tooltip === 'string' && tooltip.length > 0)) {
```
Comment (chanceaclark) [inline packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx:70]: 🍹 💅 Can we spread the negation or return it back to the original `if { return () }; return null;` route?
```suggestion
    if (typeof tooltip !== 'string' || !tooltip.length)) {
```

File: packages/big-design/src/components/Worksheet/Worksheet.tsx [hunk 5]
```diff
@@ -89,20 +101,47 @@ const InternalWorksheet = typedMemo(
       }
     }, [invalidCells, onErrors, rows]);
 
+    const getRenderedTooltip = useCallback(
+      (tooltip?: string) => {
+        if (!(typeof tooltip === 'string' && tooltip.length > 0)) {
```
Comment (chanceaclark) [inline packages/big-design/src/components/Worksheet/Worksheet.tsx:106]: 🍹 💅 Same here

**Review by chanceaclark — APPROVED** [review]

File: packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx (hunk 4 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Table/HeaderCell/HeaderCell.tsx]: Yes,  I have returned it back to its original state.

File: packages/big-design/src/components/Worksheet/Worksheet.tsx (hunk 5 above)
Comment (bc-apostoliuk) [reply packages/big-design/src/components/Worksheet/Worksheet.tsx]: Done

**Pull request merged**